qermit/postselection/postselection.py

```diff
@@ -0,0 +1,277 @@
+
+            for state in received_counts:
+                # first of all find the condensed state without ancilla bits
+                experiment_key = list(state)
+                for i in postselection_indices:
+                    del experiment_key[i]
+                experiment_key_tuple = tuple(experiment_key)
+                # set to found result, will be changed to 0 if banned
+                counts = received_counts[state]
+                new_counts.setdefault(experiment_key_tuple, 0)
+                for banned_indices in banned_bits_indices:
+                    # for each state in received_counts, we find the "sub states" that each set
+                    # of postselection bits is in
+                    # if this corresponds to a state that is "banned" we set the counts to 0
+                    if tuple([state[i] for i in banned_indices]) in banned_results:
+                        # new_counts[tuple(experiment_key)] = 0
+                        counts = 0
+                        break
+                new_counts[experiment_key_tuple] = (
+                    new_counts[experiment_key_tuple] + counts
+                )
+
+            # find remaining bits
+            remaining_bits: List[Bit] = list(result.c_bits.keys())
+            for i in postselection_indices:
+                del remaining_bits[i]
+
+            # make counter object
+            outcome_array = {
+                OutcomeArray.from_readouts([key]): val
+                for key, val in new_counts.items()
+            }
+            outcome_counts = Counter(outcome_array)
+            # add to results
+            postselected_results.append(
+                BackendResult(
+                    counts=outcome_counts,
+                    c_bits=cast(Sequence[Bit], cast(Sequence[Bit], remaining_bits)),
+                )
+            )
+        return (postselected_results,)
+
+    return MitTask(
+        _label="PostselectResults",
+        _n_in_wires=2,
+        _n_out_wires=1,
+        _method=task,
+    )
+
+
+def gen_Postselection_MitRes(
+    backend: Backend,
+    postselection_circuit: Circuit,
+    postselection_ancillas: List[Tuple[Qubit, Bit]],
+    postselection_logical_qubits: List[Qubit],
+    banned_results: Set[Tuple[bool, ...]],
+    **kwargs
+) -> MitRes:
+    """
+    Produces a MitRes object that applies Postselection techniques to experiment circuits. Each n-qubit Postselection circuit is attached to each n-qubit combination of all qubits in the experimental with end of circuit measures. Note that for k logical qubits in the experimental circuit there are n!/(n-k)! combinations to apply.
+
+    :param backend: Backend which experiments are default executed through.
+    :type backend: Backend
+    :param postselection_circuit: Circuit gadget defining some postselection scheme
+    :type postselection_circuit: Circuit
+    :param postselection_ancillas: Which Qubit in postselection_circuit are ancillas, which Bit they measure to
```

Comment:
Would it be possible to recover the 'Bit they measure to' from the circuit, rather than taking it as an input?